src/abt.sig

```diff
@@ -17,28 +41,65 @@ sig
   type metacontext = Metacontext.t
   type 'a spine = 'a Operator.Arity.Valence.Spine.t
 
+  (* The core type of the signature. This is the type of the ABTs that
+   * can be built from the given [operator]s, [variable]s, [symbol]s and
+   * [metavariable]s
+   *)
   type abt
+
+  (* An abs is an *abs*straction. It's the portion of syntax which binds
+   * variables and symbols for use in a term. It's conceptually separate
+   * from the abt type because it doesn't make sense to have a binder outside
+   * the context of some operator.
+   *)
   type abs
 
+  (* Equality on ABTs is the "right" sort of equality even though it's
+   * a little complicated. It considers terms up to
+   *  1. Alpha-varying bound variables
+   *  2. Apartness preserving renamings of bound symbols
+   *)
   structure Eq : EQ where type t = abt
 
   val freeVariables : abt -> (variable * sort) list
   val freeSymbols : abt -> (symbol * sort) list
+
+  (* This should be thought of as a "freeVariables" like function but
+   * in addition to just returning what metavariables are in scope,
+   * it also gives back information on the valence of each operator
+   * in a format that supports efficient random access
+   *)
   val metacontext : abt -> metacontext
 
   (* subst (N, x) M === [N/x]M *)
   val subst : abt * variable -> abt -> abt
 
-  (* metasubst (E, m) M === [E/m]M *)
+  (* metasubst (E, m) M === [E/m]M. Note that this does
+   * the appropriate normalization to specialize the given
```

Comment:
- [ ] Because we don't have any a way to even represent the result of a non-canonizing substitution, maybe we could say something like,

> Note that substitution of metavariables automatically instantiates the bound variables and symbols of the abstraction `E` with the operands of applications of the metavariable `m`. This operation is derived from Kevin Watkins' method of *hereditary substitution* as invented for the Concurrent Logical Framework.

(Or perhaps there is some less abstruse way to phrase the above.